File: integration-tests/iac/lambdas/python/lambda_invoker.py
import json
import os
import boto3

lambda_client = boto3.client('lambda')


def handler(event, context):
    target_function_name = os.environ['TARGET_FUNCTION_NAME']

    payload = {
        'message': 'Hello from Lambda invoker!',
        'request_id': context.aws_request_id,
    }

    response = lambda_client.invoke(
        FunctionName=target_function_name,
        InvocationType='RequestResponse',
        Payload=json.dumps(payload),
    )

    response_payload = json.loads(response['Payload'].read())
    print(f"Invoked Lambda {target_function_name}, status: {response['StatusCode']}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'invoked_function': target_function_name,
            'status_code': response['StatusCode'],
            'response': response_payload,
        }),
    }

File: integration-tests/iac/lambdas/node/lambda_invoker.mjs
import { LambdaClient, InvokeCommand } from '@aws-sdk/client-lambda';

const lambdaClient = new LambdaClient();

export async function handler(event, context) {
    const targetFunctionName = process.env.TARGET_FUNCTION_NAME;

    const payload = {
        message: 'Hello from Lambda invoker!',
        request_id: context.awsRequestId,
    };

    const response = await lambdaClient.send(new InvokeCommand({
        FunctionName: targetFunctionName,
        InvocationType: 'RequestResponse',
        Payload: JSON.stringify(payload),
    }));

    const responsePayload = JSON.parse(Buffer.from(response.Payload).toString());
    console.log(`Invoked Lambda ${targetFunctionName}, status: ${response.StatusCode}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            invoked_function: targetFunctionName,
            status_code: response.StatusCode,
            response: responsePayload,
        }),
    };
}
